pkg/configbuilder/buildconfigfile.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package configbuilder
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/klog"
+	"k8s.io/kops/util/pkg/reflectutils"
+)
+
+// ClientConnectionConfig is used by kube-scheduler to talk to the api server
+type ClientConnectionConfig struct {
+	Burst      *int32   `yaml:"burst,omitempty"`
+	Kubeconfig *string  `yaml:"kubeconfig"`
+	QPS        *float64 `yaml:"qps,omitempty"`
+}
+
+// SchedulerConfig is used to generate the config file
+type SchedulerConfig struct {
+	APIVersion         string                  `yaml:"apiVersion"`
+	Kind               string                  `yaml:"kind"`
+	BindTimeoutSeconds *int64                  `yaml:"bindTimeoutSeconds,omitempty"`
+	ClientConnection   *ClientConnectionConfig `yaml:"clientConnection,omitempty"`
+}
+
+// BuildConfigYaml reflects the options interface and extracts the parameters for the config file
+func BuildConfigYaml(options interface{}) ([]byte, error) {
```

Comment:
This is always called with a `*KubeSchedulerConfig`, so why do you declare it as `interface{}`?

Reply:
I want to generalize it later so that it can be used with others

Reply:
YAGNI applies. Later code can change it when it needs to.